core/src/main/java/com/linecorp/armeria/server/docs/DocServiceFilter.java

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Copyright 2019 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import static com.linecorp.armeria.server.docs.DocServiceBuilder.ALL_SERVICES;
+import static com.linecorp.armeria.server.docs.DocServiceBuilder.NO_SERVICE;
+import static java.util.Objects.requireNonNull;
+
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpDocServicePlugin;
+
+/**
+ * A filter which is used when building a {@link DocService}.
+ *
+ * @see DocServiceBuilder#include(DocServiceFilter)
+ * @see DocServiceBuilder#exclude(DocServiceFilter)
+ */
+@FunctionalInterface
+public interface DocServiceFilter {
+
+    /**
+     * Returns a {@link DocServiceFilter} which always returns {@code true}.
+     */
+    static DocServiceFilter allServices() {
+        return ALL_SERVICES;
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which always returns {@code false}.
+     */
+    static DocServiceFilter noService() {
```

Comment:
I wonder `allServices()` and `noService()` will ever be used by a user. Do you have a specific use case?

Reply:
Just added to use in test cases. Will move them into it. Thanks!